libs/vkd3d/state.c

```diff
@@ -2933,9 +2893,6 @@ static HRESULT d3d12_graphics_pipeline_state_create_render_pass_for_plane_mask(
         key.vk_formats[key.attachment_count++] = dsv_format;
     }
 
-    if (variant_flags & VKD3D_GRAPHICS_PIPELINE_STATIC_VARIANT_VRS_ATTACHMENT)
```

Comment:
I think we still need to cover for:
> If the current render pass instance was begun with [vkCmdBeginRenderingKHR](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#vkCmdBeginRenderingKHR) and [VkRenderingFragmentShadingRateAttachmentInfoKHR](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VkRenderingFragmentShadingRateAttachmentInfoKHR)::imageView was not [VK_NULL_HANDLE](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VK_NULL_HANDLE), the currently bound graphics pipeline must have been created with VK_PIPELINE_CREATE_RENDERING_FRAGMENT_SHADING_RATE_ATTACHMENT_BIT_KHR

If we set this unconditionally if VRS is supported, we can still use null VRS attachment.

Reply:
fixed, wasn't aware that this exists.